cmake/detect-arch.c

```diff
@@ -93,6 +93,10 @@
 #elif defined(__THW_RS6000)
     #error archfound rs6000
 
+// RISC-V
+#elif defined(__riscv)
+    #error archfound riscv
+
```

Comment:
This is oversimplified.

Reply:
Test for `__riscv` should be outer test like in the previous examples when two related preprocessor macros are checked...
Test for `__riscv_xlen` should be inner test, essentially outputting nothing if the value is neither `32` or `64`.

Reply:
Since then I've pushed another commit and currently check looks like that:
```
#elif defined(__riscv) && __riscv_xlen == 64
    #error archfound riscv64
#elif defined(__riscv) && __riscv_xlen == 32
    #error archfound riscv32
```

And it esentially do what you say, however as two checks, instead of one. If you think that it would be cleaner, I can obviously rewrite it as:
```
#elif defined(__riscv)
    #if __riscv_xlen == 64
        #error archfound riscv64
    #elif __riscv_xlen == 32
        #error archfound riscv32
```

Reply:
I think the style should be consistent with the other similar checks in the same file...

Reply:
I've force-pushed commit again. Please have a look if it is better now.